build_library/toolchain_util.sh

```diff
@@ -471,22 +463,22 @@ install_cross_libs() {
 }
 
 install_cross_rust() {
-    local cross_chost="$1"; shift
-    local emerge_flags=( "$@" --binpkg-respect-use=y --update )
-    local cbuild="$(portageq envvar CBUILD)"
-
     # may be called from either catalyst (root) or upgrade_chroot (user)
     local sudo=("env")
     if [[ $(id -u) -ne 0 ]]; then
         sudo=("sudo" "-E")
     fi
 
-    if [ "${cbuild}" = "x86_64-pc-linux-gnu" ] && [ "${cross_chost}" = "aarch64-cros-linux-gnu" ]; then
-        echo "Building Rust for arm64"
-        # If no aarch64 folder exists, try to remove any existing Rust packages.
-        [ ! -d /usr/lib/rustlib/aarch64-unknown-linux-gnu ] && ("${sudo[@]}" emerge --unmerge dev-lang/rust || true)
-        "${sudo[@]}" emerge "${emerge_flags[@]}" dev-lang/rust
-    fi
+    echo "Installing dev-lang/rust with (potentially outdated) cross targets."
+    "${sudo[@]}" emerge "${emerge_flags[@]}" --binpkg-respect-use=y --update dev-lang/rust
+
+    [[
+       -d /usr/lib/rustlib/x86_64-unknown-linux-gnu &&
+       -d /usr/lib/rustlib/aarch64-unknown-linux-gnu
+    ]] && return
+
+    echo "Rebuilding dev-lang/rust with updated cross targets."
+    "${sudo[@]}" emerge "${emerge_flags[@]}" --usepkg=n dev-lang/rust
```

Comment:
There seem to be a possible scenario where we build rust twice?

Reply:
Yes, even before, it would install Rust from a binary package and then rebuild it if it didn't have the necessary cross targets. The previous logic was broken though, as it would do this once per cross target instead of just once. It's not like crossdev, the single package covers all the targets.